### ISSUE TITLE
Array and jwt matchers fail with invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by romangr** [SpectoLabs/hoverfly-java#284]
## Description of the bug

I use `hoverfly-java` for my tests and whenever I try to use array/jwt matcher Hoverfly can't serve a response. It doesn't affect other requests if they don't match with the array/jwt matchers even if such matchers are present in the config. This also can be reproduced on the binary from `hoverfly-java`.

## Steps to reproduce the issue

1. Prepare config
```json
{
    "data": {
        "pairs": [
            {
                "response": {
                    "status": 200,
                    "body": "{\"test\": true}",
                    "encodedBody": false,
                    "headers": {
                        "Content-Type": [
                            "application/json; charset=utf-8"
                        ]
                    }
                },
                "request": {
                    "path": [
                        {
                            "matcher": "exact",
                            "value": "/v1/entity"
                        }
                    ],
                    "query": {
                        "ids": [
                            {
                                "matcher": "array",
                                "config": {
                                    "ignoreUnknown": false,
                                    "ignoreOrder": true,
                                    "ignoreOccurrences": false
                                },
                                "value": [
                                    "fb34bdee-31d3-46a0-8cc8-b214e293b486",
                                    "9ee6e7e3-686c-4944-982f-3301c111cd89"
                                ]
                            }
                        ]
                    },
                    "method": [
                        {
                            "matcher": "exact",
                            "value": "GET"
                        }
                    ],
                    "scheme": [
                        {
                            "matcher": "exact",
                            "value": "http"
                        }
                    ]
                }
            }
        ],
        "globalActions": {
            "delays": []
        }
    },
    "meta": {
        "schemaVersion": "v5"
    }
}
``` 
2. Run hoverfly `./hoverfly_OSX_arm64 -webserver -import test.json`
3. Make a request that matches this
```
curl -G -d 'ids=9ee6e7e3-686c-4944-982f-3301c111cd89' -d 'ids=fb34bdee-31d3-46a0-8cc8-b214e293b486' --location 'http://localhost:8500/v1/entity'
```
4. See an error in the logs

### Observed result

Hoverfly logs:
```
2023/08/23 17:30:40 http: panic serving 127.0.0.1:55445: runtime error: invalid memory address or nil pointer dereference
goroutine 5 [running]:
net/http.(*conn).serve.func1(0x14000114000)
	/usr/local/go/src/net/http/server.go:1805 +0x108
panic(0x1032163c0, 0x1037fcaa0)
	/usr/local/go/src/runtime/panic.go:971 +0x3f4
github.com/SpectoLabs/hoverfly/core/matching.FieldMatcher(0x140004860e0, 0x1, 0x1, 0x14000072083, 0x49, 0x14000072083)
	/home/circleci/hoverfly/core/matching/field_matcher.go:18 +0xc4
github.com/SpectoLabs/hoverfly/core/matching.QueryMatching(0x14000486080, 0x1, 0x1, 0x140004860a0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x140004860c0, ...)
	/home/circleci/hoverfly/core/matching/query_matching.go:48 +0x26c
github.com/SpectoLabs/hoverfly/core/matching.MatchingStrategyRunner(0x14000072074, 0xa, 0x14000072070, 0x3, 0x1400003e0b0, 0xe, 0x103018ee7, 0x4, 0x14000108390, 0x0, ...)
	/home/circleci/hoverfly/core/matching/matching_strategy_runner.go:38 +0x320
github.com/SpectoLabs/hoverfly/core/matching.Match(0x10302caf9, 0x9, 0x14000072074, 0xa, 0x14000072070, 0x3, 0x1400003e0b0, 0xe, 0x103018ee7, 0x4, ...)
	/home/circleci/hoverfly/core/matching/matcher.go:12 +0x134
github.com/SpectoLabs/hoverfly/core.(*Hoverfly).GetResponse(0x14000377cc0, 0x14000072074, 0xa, 0x14000072070, 0x3, 0x1400003e0b0, 0xe, 0x103018ee7, 0x4, 0x14000108390, ...)
	/home/circleci/hoverfly/core/hoverfly_funcs.go:65 +0x468
github.com/SpectoLabs/hoverfly/core/modes.SimulateMode.Process(0x1032f7370, 0x14000377cc0, 0x10302caf9, 0x9, 0x14000120000, 0x14000072074, 0xa, 0x14000072070, 0x3, 0x1400003e0b0, ...)
	/home/circleci/hoverfly/core/modes/simulate_mode.go:44 +0x84
github.com/SpectoLabs/hoverfly/core.(*Hoverfly).processRequest(0x14000377cc0, 0x14000120000, 0x103827ac0)
	/home/circleci/hoverfly/core/hoverfly.go:199 +0x11c
github.com/SpectoLabs/hoverfly/core.NewWebserverProxy.func1(0x1032fcdf0, 0x14000126000, 0x14000120000)
	/home/circleci/hoverfly/core/proxy.go:107 +0x84
net/http.HandlerFunc.ServeHTTP(0x140002e5cc0, 0x1032fcdf0, 0x14000126000, 0x14000120000)
	/usr/local/go/src/net/http/server.go:2050 +0x40
github.com/SpectoLabs/goproxy.(*ProxyHttpServer).ServeHTTP(0x14000488000, 0x1032fcdf0, 0x14000126000, 0x14000120000)
	/home/circleci/hoverfly/vendor/github.com/SpectoLabs/goproxy/proxy.go:135 +0xa20
net/http.serverHandler.ServeHTTP(0x140001ff260, 0x1032fcdf0, 0x14000126000, 0x14000120000)
	/usr/local/go/src/net/http/server.go:2868 +0xbc
net/http.(*conn).serve(0x14000114000, 0x1032ff080, 0x14000030140)
	/usr/local/go/src/net/http/server.go:1933 +0x710
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2994 +0x308
```

### Expected result

Response from request-response pair is served by Hoverfly

## Additional relevant information
1. Hoverfly version: the version from io.specto:hoverfly-java:0.14.4 - v1.4.0
2. I use M1 Pro and tested with `hoverfly_OSX_arm64` and `hoverfly_OSX_amd64`, both fail

**Comment by romangr** [comment]
@tommysitu Does it make sense to supply a new version of hoverfly in this case?

**Comment by romangr** [comment]
Thank you for merging the PR, it'd be great if you can trigger a release of the new version

**Comment by tommysitu** [comment]
Have you tried out the latest snapshot version? If this works for you, I will do a release.

**Comment by romangr** [comment]
Just checked, now array matcher works as expected

**Comment by tommysitu** [comment]
@romangr thanks, I will give it a go today.

**Comment by tommysitu** [comment]
your fix should be available in v0.15.0 :)

**Comment by romangr** [comment]
Cool, thank you!
